### PR TITLE
Add a legacy initializer for animation traits.

### PR DESCRIPTION
**Pull request by jverkoey**
This will allow current usage of the legacy C APIs to migrate to the newer Objective-C APIs more easily.

### CODE REVIEW

File: src/MDMAnimationTraits.h [hunk 1]
```diff
@@ -100,6 +101,13 @@ MDM_SUBCLASSING_RESTRICTED
                            repetition:(nullable id<MDMRepetitionTraits>)repetition
     NS_DESIGNATED_INITIALIZER;
 
+/**
+ Initializes the instance with the provided legacy C-style animation trait structure.
+
+ @param timing A legacy C-style representation of animation traits.
+ */
+- (nonnull instancetype)initWithMotionTiming:(MDMMotionTiming)timing;
```
Comment (romoore) [inline src/MDMAnimationTraits.h]: Should this be in a category for easy removal later? Or at least to keep track of the "transitional" code?

File: src/MDMAnimationTraits.m [hunk 2]
```diff
@@ -80,6 +82,50 @@ - (instancetype)initWithDelay:(NSTimeInterval)delay
...
+      timingCurve = [CAMediaTimingFunction functionWithControlPoints:(float)timing.curve.data[0]
+                                                                    :(float)timing.curve.data[1]
+                                                                    :(float)timing.curve.data[2]
+                                                                    :(float)timing.curve.data[3]];
+      break;
+    case MDMMotionCurveTypeSpring:
+      timingCurve = [[MDMSpringTimingCurve alloc] initWithMass:timing.curve.data[MDMSpringMotionCurveDataIndexMass]
+                                                       tension:timing.curve.data[MDMSpringMotionCurveDataIndexTension]
+                                                      friction:timing.curve.data[MDMSpringMotionCurveDataIndexFriction]
+                                               initialVelocity:timing.curve.data[MDMSpringMotionCurveDataIndexInitialVelocity]];
```
Comment (romoore) [inline src/MDMAnimationTraits.m]: Nit: Is this line too long?

**Review by romoore — APPROVED** [review]
Unit tests?

File: src/MDMAnimationTraits.h (hunk 1 above)
Comment (jverkoey) [reply src/MDMAnimationTraits.h]: Good call. Moved.

File: src/MDMAnimationTraits.m (hunk 2 above)
Comment (jverkoey) [reply src/MDMAnimationTraits.m]: Fixed within reason.

**Pull request merged**